### DISCUSSION TITLE
issue: Pattern validation doesn't work with react-input-mask

### DISCUSSION THREAD

**Discussion by nashyvan** [react-hook-form/react-hook-form#9865] [Q&A] [answered]
### Version Number

7.43.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-form-validation-with-input-mask-qcbikz

### Steps to reproduce

Without react-input-mask pattern validation works great:
https://codesandbox.io/s/react-form-validation-s3fo1m
With react-input-mask pattern validation doesn't work:
https://codesandbox.io/s/react-form-validation-with-input-mask-qcbikz

### Expected behaviour

validation must work

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment, marked as answer]
`InputMask` handles the input value internally passing the already formatted value to the RHF `onChange` handler, i.e. `+38 (___) ___-__-__`, which is a valid string against the `RegEx` you've provided

**Reply by nashyvan** [reply]
so this is problem with react-input-mask library?

**Reply by nashyvan** [reply]
`maskChar={null}` solve this issue for react-input-mask v2
`maskPlaceholder={null}` solve this issue for react-input-mask v3 
thank you

**Reply by Moshyfawn** [reply]
It's not a **problem** but the way the `InputMask` input works.

And you're right, `maskChar={null}` emits the desired behaviour for your use-case 👍
